app/controllers/api/base_controller.rb

```diff
@@ -43,6 +43,32 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def response_with_warning(response)
```

Comment:
I think we can remove this as well. Just use `response_with_mfa_warning` directly in controller. That seems most transparent to me.